### PR TITLE
FIX: Implement missing __rtruediv__ operator for Explanation objects

### LINKED ISSUE #4785: BUG: Missing Explanation.__rtruediv__ causes TypeError for scalar / explanation operations
### Issue Description

The `shap.Explanation` class implements a full suite of arithmetic magic methods to support algebraic manipulation of SHAP values. It correctly implements right-hand operators for addition and subtraction (`__radd__`, `__rsub__`), but is missing `__rtruediv__`.

This means that while `shap_values / 2.0` works, the mathematically equivalent inverse operation `1.0 / shap_values` raises a `TypeError`, making certain normalization patterns impossible without workarounds and breaking algebraic symmetry with the existing right-hand operator support.

This is the same class of missing operator as identified in #4709 (`__rsub__`), where a right-hand operator was either absent or incorrectly implemented on the `Explanation` class.

### Minimal Reproducible Example

```python
import numpy as np
import shap

exp = shap.Explanation(
    values=np.array([1.0, 2.0, 4.0]),
    base_values=0.0,
    data=np.array([0.1, 0.2, 0.3]),
    feature_names=["f0", "f1", "f2"]
)

# These all work correctly:
print(exp / 2.0)   # __truediv__  ✓
print(1.0 * exp)   # __rmul__     ✓
print(1.0 + exp)   # __radd__     ✓

# This crashes:
print(1.0 / exp)   # __rtruediv__ — missing
```

### Traceback

```shell
TypeError: unsupported operand type(s) for /: 'float' and 'Explanation'
```

### Expected Behavior

`1.0 / exp` should return a new `Explanation` object where each value in `exp.values` is used as the denominator:

```python
values: [1/1.0, 1/2.0, 1/4.0] = [1.0, 0.5, 0.25]
```

This is mathematically consistent with how `__rmul__` and `__radd__` already behave on the same class.

**Proposed fix** — add to `shap/_explanation.py`:

```python
def __rtruediv__(self, other: Explanation | npt.NDArray[Any] | float | int) -> Explanation:
    return self._apply_binary_operator(other, lambda a, b: b / a, "__rtruediv__")
```

Note: passing `operator.truediv` directly would repeat the same operand-order bug present in `__rsub__` before #4709, computing `self / other` instead of the correct `other / self`. The operands must be explicitly reversed.

### Bug report checklist

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the [latest release](https://github.com/shap/shap/releases) of shap.
- [x] I have confirmed this bug exists on the [master branch](https://github.com/shap/shap/blob/master/CONTRIBUTING.md#installing-from-the-master-branch) of shap.
- [x] I'd be interested in making a PR to fix this bug

### Installed Versions

shap: current master
python: 3.11
numpy: 2.x

### PR DESCRIPTION
**Pull request by GobinathMaruthalingam**
Fixes #4785.

### Description
The `Explanation` class implements `__truediv__` but is missing `__rtruediv__`. This causes `1.0 / shap_values` to crash with a `TypeError`, breaking algebraic symmetry with the existing right-hand operator support (`__radd__`, `__rmul__`).

This PR adds `__rtruediv__` with explicitly reversed operands (`lambda a, b: b / a`) to prevent repeating the same operand-order bug seen in `__rsub__` before #4709.

### Tests added
- `float / explanation` — core behaviour
- `int / explanation` — non-float scalar edge case
- Operand-order regression guard confirming `other / self` is computed, not `self / other`

### Note on NumPy array behaviour
A test for `numpy_array / explanation` was considered but omitted. When the left operand is an ndarray, NumPy's own `__truediv__` takes precedence and `Explanation.__rtruediv__` is never invoked. This matches the existing behaviour across all other right-hand operators in the class (`__radd__`, `__rmul__`) and is not specific to this PR.

### Checklist
- [x] Unit tests added covering float and int divisors
- [x] Operand-order regression guard included
- [x] All existing tests pass locally

### CODE REVIEW

File: tests/test_explanation.py [hunk 1]
```diff
@@ -245,3 +245,28 @@ def test_cohorts_generation_with_one_feature():
     cohorts = exp.cohorts(3)
     assert isinstance(cohorts, shap.Cohorts)
     assert len(cohorts.cohorts) == 3
+
+
+def test_explanation_rtruediv():
+    """Checks that __rtruediv__ correctly computes other / self, not self / other."""
+    import numpy as np
+
+    import shap
```
Comment (SuMayaBee) [inline tests/test_explanation.py]: We don't need those imports, numpy and shap are already imported at the top of the file, these imports inside the function body are redundant.

Comment (GobinathMaruthalingam) [reply tests/test_explanation.py]: Oops, didn't see that
Removed in **bb2cef96**, dropped the redundant imports
